tobler/util/util.py

```diff
@@ -102,14 +106,76 @@ def project_gdf(gdf, to_crs=None, to_latlong=False):
 
     # calculate the centroid of the union of all the geometries in the
     # GeoDataFrame
-    avg_longitude = gdf['geometry'].unary_union.centroid.x
+    avg_longitude = gdf["geometry"].unary_union.centroid.x
 
     # calculate the UTM zone from this avg longitude and define the UTM
     # CRS to project
-    utm_zone = int(math.floor((avg_longitude + 180) / 6.) + 1)
-    utm_crs = '+proj=utm +zone={} +ellps=WGS84 +datum=WGS84 +units=m +no_defs'.format(utm_zone)
+    utm_zone = int(math.floor((avg_longitude + 180) / 6.0) + 1)
+    utm_crs = "+proj=utm +zone={} +ellps=WGS84 +datum=WGS84 +units=m +no_defs".format(
+        utm_zone
+    )
 
     # project the GeoDataFrame to the UTM CRS
     projected_gdf = gdf.to_crs(utm_crs)
 
     return projected_gdf
+
+
+def hexify(source, resolution=6, clip=False):
+    """Generate a hexgrid geodataframe that covers the face of a source geodataframe.
+
+    Parameters
+    ----------
+    source : geopandas.GeoDataFrame
+        GeoDataFrame to transform into a hexagonal grid
+    resolution : int, optional
+        resolution of output h3 hexgrid. 
+        See <https://h3geo.org/docs/core-library/restable> for more information
+    clip : bool, optional
+        if True, hexagons are clipped to the precise boundary of the source gdf. Otherwise,
+        heaxgons along the boundary will be left intact.
+
+    Returns
+    -------
+    geopandas.GeoDataFrame
+        a GeoDataFrame whose rows comprise a hexagonal h3 grid. 
+    """
+    try:
+        from h3 import h3
+    except ImportError:
+        raise ImportError(
+            "This function requires the `h3` library. "
+            "You can install it with `conda install h3` or "
+            "`pip install h3`"
+        )
+    orig_crs = source.crs.to_string()
+
+    if not source.crs.name == "WGS 84":
+        source = source.copy().to_crs(4326)
```

Comment:
```suggestion
    orig_crs = source.crs

    if not source.crs.is_geographic:
        source = source.to_crs(4326)
```
String can be tricky for custom projections and is not needed.

Reply:
cool, thanks. I originally had it that way, but i wasnt sure if `orig_crs=source.crs` might just create a pointer rather than stash the original pyproj object.

is it cool to modify the input by reprojecting without copying first?

Reply:
>  i wasnt sure if orig_crs=source.crs might just create a pointer rather than stash the original pyproj object.

I am not sure about this either, but since we create new `source` object below it should preserve original (worth testing though)

> is it cool to modify the input by reprojecting without copying first?

`to_crs()` does not work in place (unless you specify `inplace=True`), so it should be okay.